src/Microsoft.AspNetCore.SystemWebAdapters/Configuration/HttpCapabilitiesBase.cs

```diff
@@ -53,6 +53,14 @@ private IHttpBrowserCapabilityFeature Capability
 
     public string? this[string key] => Capability[key];
 
+    public Version? EcmaScriptVersion => new Version(Capability["ecmascriptversion"] ?? string.Empty);
```

Comment:
```suggestion
    public Version? EcmaScriptVersion =>Capability["ecmascriptversion"] ?? is {} v ? new Version(v) : null;
```

You'll probably want to make a helper method to do that. However, verify what .NET Framework did here - did it return a null (I'm assuming that based on your nullable annotation), or did it return a version of something?

Reply:
https://referencesource.microsoft.com/#mscorlib/system/version.cs,354  and here https://referencesource.microsoft.com/#mscorlib/system/version.cs,376 <= eventually if the version is null or empty or invalid it throws exception.  So I am making the API as not nullable to stick with framework and adding helper to handle as below. Let me know if looks below, I will submit the next revision.

```c#
    public Version EcmaScriptVersion => GetVersion("ecmascriptversion");
   
    private Version GetVersion(string key)
    {
        var result = Capability[key];
        if (result is not null)
        {
            return new Version(result);
        }
        else
        {
            throw new ArgumentNullException(nameof(key));
        }
    }
```

Reply:
If that's what framework is doing, that sounds good to me